Rubberduck.CodeAnalysis/Inspections/Concrete/UnreachableCaseInspection/ParseTreeExpressionEvaluator.cs

```diff
@@ -61,32 +66,79 @@ public class ParseTreeExpressionEvaluator : IParseTreeExpressionEvaluator
+            };
+
         }
 
         public IParseTreeValue Evaluate(IParseTreeValue LHS, IParseTreeValue RHS, string opSymbol)
         {
             var isMathOp = MathOpsBinary.ContainsKey(opSymbol);
             var isLogicOp = LogicOpsBinary.ContainsKey(opSymbol);
-            Debug.Assert(isMathOp || isLogicOp);
+            var isBinaryStringOp = opSymbol.Equals(Tokens.Like) || opSymbol.Equals(_ampersand);
+            Debug.Assert(IsSupportedSymbol(opSymbol));
 
-            var opResultTypeName = isMathOp ? DetermineMathResultType(LHS, RHS) : Tokens.Boolean;
-            var operands = PrepareOperands(new IParseTreeValue[] { LHS, RHS });
+            (string lhsTypeName, double lhsValue) = PrepareOperand(LHS);
+            (string rhsTypeName, double rhsValue) = PrepareOperand(RHS);
 
-            if (operands.Count == 2)
+            if (!lhsTypeName.Equals(string.Empty) && !rhsTypeName.Equals(string.Empty))
             {
                 if (isMathOp)
                 {
-                    var mathResult = MathOpsBinary[opSymbol](operands[0], operands[1]);
-                    return _valueFactory.Create(mathResult.ToString(CultureInfo.InvariantCulture), opResultTypeName);
+                    var mathResult = MathOpsBinary[opSymbol](lhsValue, rhsValue);
+                    return _valueFactory.Create(mathResult.ToString(CultureInfo.InvariantCulture), DetermineMathResultType(lhsTypeName, rhsTypeName));
+                }
+                else if (isLogicOp)
+                {
+                    if (lhsTypeName.Equals(Tokens.String) && rhsTypeName.Equals(Tokens.String))
+                    {
+                        if (LogicOpsString.ContainsKey(opSymbol))
+                        {
+                            var stringLogicResult = LogicOpsString[opSymbol](LHS.ValueText, RHS.ValueText);
+                            return _valueFactory.Create(stringLogicResult.ToString(CultureInfo.InvariantCulture), Tokens.Boolean);
+                        }
+                        //Store invalid string logic op result as variable predicate
+                        return _valueFactory.Create($"{LHS.ValueText} {opSymbol} {RHS.ValueText}", Tokens.Boolean);
+                    }
+                    var logicResult = LogicOpsBinary[opSymbol](lhsValue, rhsValue);
+                    return _valueFactory.Create(logicResult.ToString(CultureInfo.InvariantCulture), Tokens.Boolean);
+                }
+            }
+
+            if (isBinaryStringOp)
+            {
+                if (opSymbol.Equals(_ampersand))
+                {
+                    var concatResult = $"{Concat(LHS.ValueText, RHS.ValueText)}";
+                    return _valueFactory.Create(concatResult, Tokens.String);
+                }
+
+                if (opSymbol.Contains(Tokens.Like))
+                {
+                    if (LHS.ParsesToConstantValue && RHS.ParsesToConstantValue)
```

Comment:
Good points.  The combination of `foo Like "*"` and `Not foo Like "*"` needs to result in setting values True and False respectively in the `ExpressionFilter<T>`.  Any subsequent relational operator cases ( `Like` or otherwise) would then be flagged as unreachable.  (Note: if the `Select Case Context` is of a type other than `Boolean`, then `Case Else` is still in-play).  I have this fix ready to go on my local machine.

What I realize is missing, is the use case where the `SelectExpressionContext` resolves to a constant value (as in `Select Case True`).  In that use case, the inspection could determine which `CaseClause` (if any) are equal, and flag the rest as unreachable.  Currently, the inspection does not consider the `SelectExpressionContext` value, only its Type.  The constant `SelectExpressionContext` use case is probably rare except for `Select Case True/False` - but handling the `Boolean` case would handle the rest.

As to `foo Like "?*"` making `"foo Like a*"` unreachable - I agree.  I did not wade into the task of evaluating Like patterns for superset/subset effects.  That said, Like patterns that have a variable LHS are retained (indexed by the LHS variable).  The associated Like patterns are evaluated/filtered for equality (finds duplicates) before storing.  The 'TODO' for adding superset/subset evaluations would go there.  I will poke at this enhancement a little...on the surface it looks difficult.

Reply:
Updated per above comments.  
1.  Handles use case where a `SelectExpressionContext` resolves to a constant value.  
2.  Set up for `Like` pattern filtering.